docs/configuration/testing-behavior.rst

```diff
@@ -111,3 +111,75 @@ When at least one WARN message is present, the function passed to ``on_warning_c
     If warnings that are not associated with tests occur (e.g. freshness warnings), they will still trigger the
     ``on_warning_callback`` method above. However, these warnings will not be included in the ``test_names`` and
     ``test_results`` context variables, which are specific to test-related warnings.
+
+
+Tests with Multiple Parents
+---------------------------
+
+It is common for dbt projects to define tests that rely on multiple upstream models, snapshots or seeds.
+By default, Cosmos will attempt to run these tests using the behavior defined using ``test_behavior`` as previously explained.
+
+As an example, if there is a test that depends on multiple models (``model_a`` and ``combined_model``), and the DAG uses
+``TestBehavior.AFTER_EACH``, Cosmos will attempt to run this test twice after each model run.
+
+While the standard behavior of Cosmos works for many cases, there are a few scenarios when the test fails unless both models
+run. To overcome this issue, starting in Cosmos 1.8.2, we introduced the parameter
+``should_detach_multiple_parents_tests`` in ``RenderConfig``. By default, it is ``False``. If it is set to ``True`` and
+``TestBehavior` is ``AFTER_EACH`` or ``BUILD``, Cosmos will identify all the test nodes that depend on multiple parents
+and will create a standalone test task for each of them.
+
+Cosmos will attempt to name this task after the test original name. Since some test names can be very long (over 250 characters)
+and Airflow does not support IDs longer than 250 characters; Cosmos will name them "detached_0_test", incrementing
+0 as needed.
```

Comment:
```suggestion
Cosmos will attempt to name this task after the test's original name. However, since some test names can exceed 250 characters and Airflow does not support IDs longer than this limit, Cosmos will assign names like “detached_0_test,” incrementing the number as needed.
```